### ISSUE TITLE
tedge cert command to create a certificate signing request

### ISSUE THREAD

**Issue by reubenmiller** [thin-edge/thin-edge.io#2760]
**Is your feature request related to a problem? Please describe.**

If the user wants to use a device certificate which is signed by a Certificate Authority (CA), then they have to use other tooling to generate the Certificate Signing Request (CSR) which is used by the CA to generate the device certificate (in most instances this is openssl). The reliance of additional tooling means that users have to have this tooling installed, and also know how to use it which can be daunting for the less experienced users.

**Describe the solution you'd like**

A new `tedge cert` command to generate a Certificate Signing Request file which the user can then use to generate the public device certificate (via their CA).

For example, the command could look something like:

```sh
# Create a new private key (if it does not already exist)
tedge cert create-csr --device-id mydevice001 > mydevice.csr

# Re-use the existing device-id from an already existing certificate
tedge cert create-csr > mydevice.csr
```

Where:
* A private key will be created automatically if it does not already exist
* If a public key already exists, then re-use the device-id from the CommonName field of the public certificate (similar to what the `tedge cert renew` command does)

**Describe alternatives you've considered**

**Additional context**

**Comment by Ruadhri17** [comment]
Resolved by #2783

Ticket introduces new command `tedge cert create-csr` that create a Certificate Signing Request under directory that can be set with `tedge config set device.csr_path` (by default it is `/etc/tedge/device-certs/tedge.csr`). User can also provide output path as command argument with `--output-path`. 

More detailed description, how a CSR is created, can be found [here](https://github.com/thin-edge/thin-edge.io/blob/main/docs/src/operate/security/certificate-signing-request.md). 

System test was created and can be found [here](https://github.com/thin-edge/thin-edge.io/blob/main/tests/RobotFramework/tests/tedge/certificate_signing_request.robot).

**Comment by gligorisaev** [comment]
QA has thoroughly checked the feature and here are the results:

- [x] Test for ticket exists in the test suite.
- tests/RobotFramework/tests/tedge/certificate_signing_request.robot
- [x] QA has tested the function and it's functioning according
